src/gpuhunt/providers/oci.py

```diff
@@ -0,0 +1,229 @@
+    part_number: str
+    qty: int
+
+    class Config:
+        alias_generator = to_camel_case
+
+
+class CostEstimatorShape(BaseModel):
+    name: str
+    hidden: bool
+    status: str
+    bundle_memory_qty: int
+    gpu_qty: Optional[int]
+    gpu_memory_qty: Optional[int]
+    processor_type: CostEstimatorTypeField
+    shape_type: CostEstimatorTypeField
+    sub_type: CostEstimatorTypeField
+    products: List[CostEstimatorShapeProduct]
+
+    class Config:
+        alias_generator = to_camel_case
+
+    def is_arm_cpu(self):
+        is_ampere_gpu = self.sub_type.value == "gpu" and (
+            "GPU4" in self.name or "GPU.A10" in self.name
+        )
+        # the data says A10 and A100 GPU instances are ARM, but they are not
+        return self.processor_type.value == "arm" and not is_ampere_gpu
+
+    def get_gpu_unit_memory_gb(self) -> Optional[float]:
+        if self.gpu_memory_qty and self.gpu_qty:
+            return self.gpu_memory_qty / self.gpu_qty
+        return None
+
+
+class CostEstimatorShapeList(BaseModel):
+    items: List[CostEstimatorShape]
+
+
+class CostEstimatorPrice(BaseModel):
+    model: str
+    value: float
+
+
+class CostEstimatorPriceLocalization(BaseModel):
+    currency_code: str
+    prices: List[CostEstimatorPrice]
+
+    class Config:
+        alias_generator = to_camel_case
+
+
+class CostEstimatorProduct(BaseModel):
+    part_number: str
+    billing_model: str
+    price_type: Annotated[str, Field(alias="pricetype")]
+    currency_code_localizations: List[CostEstimatorPriceLocalization]
+
+    class Config:
+        alias_generator = to_camel_case
+
+    def find_price_l10n(self, currency_code: str) -> Optional[CostEstimatorPriceLocalization]:
+        return next(
+            filter(
+                lambda price: price.currency_code == currency_code,
+                self.currency_code_localizations,
+            ),
+            None,
+        )
+
+
+class CostEstimatorProductList(BaseModel):
+    items: List[CostEstimatorProduct]
+
+    def find(self, part_number: str) -> Optional[CostEstimatorProduct]:
+        return next(filter(lambda product: product.part_number == part_number, self.items), None)
+
+
+class CostEstimator:
+    def __init__(self):
+        self.session = Session()
+
+    def get_shapes(self) -> CostEstimatorShapeList:
+        return self._get("shapes.json", CostEstimatorShapeList)
+
+    def get_products(self) -> CostEstimatorProductList:
+        return self._get("products.json", CostEstimatorProductList)
+
+    def _get(self, resource: str, ResponseModel: Type[BaseModel]):
+        url = COST_ESTIMATOR_URL_TEMPLATE.format(resource=resource)
+        resp = self.session.get(url, timeout=COST_ESTIMATOR_REQUEST_TIMEOUT)
+        resp.raise_for_status()
+        return ResponseModel.parse_raw(resp.content)
+
+
+def get_gpu_name(shape_name: str) -> Optional[str]:
+    parts = re.split(r"[\.-]", shape_name)
+    parts = list(map(str.upper, parts))
```

Comment:
Let's reduce it by 1 line )
```suggestion
    parts = re.split(r"[\.-]", shape_name.upper())
```